session/session.go

```diff
@@ -540,7 +541,31 @@ func (s *session) CommitTxn(ctx context.Context) error {
 
 	var commitDetail *execdetails.CommitDetails
 	ctx = context.WithValue(ctx, execdetails.CommitDetailCtxKey, &commitDetail)
-	err := s.doCommitWithRetry(ctx)
+	var err error
```

Comment:
I'm not sure if it's fine to check the constraint in this function. I'll double-check it.